packages/cells/src/widget.ts

```diff
@@ -1325,8 +1326,19 @@ export abstract class AttachmentsCell extends Cell {
       const mimeType = matches[1];
       const encodedData = matches[3];
       const bundle: nbformat.IMimeBundle = { [mimeType]: encodedData };
-      this.model.attachments.set(blob.name, bundle);
-      this.updateCellSourceWithAttachment(blob.name);
+
+      // Find an unused URI
+      let URI = encodeURI(blob.name)
+      let URI_array = URI.split(/(\.)(?=[^\.]+$)/);
```

Comment:
You could also use `.lastIndexOf`, which may be faster and clearer than a regex? Since you have to splice below anyway, maybe it's not any longer as well?

Reply:
I ended up using the regex because I was struggling with slicing when there was no `.` present and `lastIndexOf` returns -1.
I suppose we could do
```typescript
      let lastIndex = URI.lastIndexOf('.');
      let URI_array = [];
      if (lastIndex != -1){
        URI_array = [URI.substring(0,lastIndex),'', URI.substring(lastIndex)];
      } else {
        URI_array = [URI, ''];
      }
```

Reply:
You don't need to initialize to the empty array, just `let URI_array` would work. Or perhaps using the ternary is clear enough and shorter:

```typescript
const lastIndex = URI.lastIndexOf('.');
let URI_array = lastIndex !== -1 ? [URI.substring(0,lastIndex),'', URI.substring(lastIndex)] : [URI, ''];
```

Reply:
Though your other arguments convince me that we should just use a UUID and not worry about this.

Reply:
Switched to uuid approach.